source/common/tracing/http_tracer_impl.cc

```diff
@@ -158,6 +158,8 @@ void HttpTracerUtility::finalizeSpan(Span& active_span, const Http::HeaderMap& r
 HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& local_info)
     : driver_(std::move(driver)), local_info_(local_info) {}
 
+void HttpTracerImpl::installDriver(DriverPtr&& driver) { driver_ = std::move(driver); }
```

Comment:
No one should start a span between the time we make a tracer, and install the driver. I would remove driver from the constructor, and allow it to be null. Then in this function I would ASSERT(driver_ == nullptr);